src/Components/PaperEditor/index.js

```diff
@@ -60,20 +72,56 @@ const PaperEditor = (props) => {
       }
     };
 
+    if (!transcripts) {
+      getPaperEdit();
+    }
+
+    return () => { };
+  }, [ PaperEdits, papereditId, transcripts ]);
+
+  useEffect(() => {
     const getLabels = async () => {
       try {
-        Labels.collectionRef.onSnapshot((snapshot) => {
-          setLabels(
+        await Labels.collectionRef.onSnapshot((snapshot) => {
+          const tempLabels = snapshot.docs.map((doc) => {
+            return { ...doc.data(), id: doc.id, display: true };
+          });
+          setLabels(tempLabels);
+        });
+      } catch (error) {
+        console.error('Error getting labels: ', error);
+      }
+    };
+
+    if (!transcripts) {
+      getLabels();
+    }
+
+    return () => {};
+  }, [ Labels.collectionRef, transcripts ]);
+
+  useEffect(() => {
+    const getAnnotations = async () => {
+      try {
+        Annotations.collectionRef.onSnapshot((snapshot) => {
+          setAnnotations(
             snapshot.docs.map((doc) => {
               return { ...doc.data(), id: doc.id, display: true };
             })
           );
         });
       } catch (error) {
-        console.error('Error getting labels: ', error);
+        console.error('Error getting annotations: ', error);
       }
     };
+    if (!transcripts) {
```

Comment:
probably worth setting annotations to `null` to begin with, so you can use annotations instead to check if you're fetching for annotations. This way you also don't have to subscribe to transcripts in the useEffect.

Reply:
Changing this causes the view not to load; the original code was also written in a way where transcripts were checked for each of these values. I'm not sure why this happens and your change makes sense ... but it causes the view to try to load before the other props have been grabbed from Firebase.